### PR TITLE
Week4 - Added TSP + Maps API and Charts API

### PR DESCRIPTION
**Pull request by diego-escobedo**

### CODE REVIEW

File: portfolio/src/main/webapp/css/map.css [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#map {
+  border: thin solid black;
+  height: 600px;
+  width: 600px;
+}
+
+li {
+    margin: 0;
```
Comment (blhtkd) [inline portfolio/src/main/webapp/css/map.css:8]: spacing is off here and below

File: portfolio/src/main/webapp/js/charts.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+google.charts.load('current', {'packages':['corechart']});
+google.charts.setOnLoadCallback(drawChart);
+
+/** Creates a chart and adds it to the page. */
+function drawChart() {
+  const data = new google.visualization.DataTable();
+  data.addColumn('string', 'Animal');
+  data.addColumn('number', 'Count');
+        data.addRows([
```
Comment (blhtkd) [inline portfolio/src/main/webapp/js/charts.js:23]: indentation

File: portfolio/src/main/webapp/js/twoopt.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                break;
+            }
+        }
+        if (better_found === true) {
+                continue;
+            }
+        else {
+            break;
+        }
+    }
```
Comment (blhtkd) [inline portfolio/src/main/webapp/js/twoopt.js:33]: Can you try to improve this algorithm, specifically all the break and continue checks would not be allowed in the google codebase. also formatting is a little off.

File: portfolio/src/main/webapp/js/twoopt.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    return existing_route;
+}
+
+function calculateTotalDistance(route,distmat) {
+
```
Comment (blhtkd) [inline portfolio/src/main/webapp/js/twoopt.js:38]: delete empty line

File: portfolio/src/main/webapp/js/twoopt.js (hunk 3 above)
Comment (diego-escobedo) [reply portfolio/src/main/webapp/js/twoopt.js]: the original algorithm has a goto so I need 2 breaks to exit the nested for loops, but I removed an unnecessary break/continue at the end

**Pull request merged**